src/rust/engine/process_execution/src/local.rs

```diff
@@ -380,16 +305,23 @@ impl CapturedWorkdir for CommandRunner {
     workdir_path: &'b Path,
     req: Process,
     _context: Context,
+    exclusive_spawn: bool,
   ) -> Result<BoxStream<'c, Result<ChildOutput, String>>, String> {
-    StreamedHermeticCommand::new(&req.argv[0])
+    let mut command = StreamedHermeticCommand::new(&req.argv[0]);
+    command
       .args(&req.argv[1..])
       .current_dir(if let Some(ref working_directory) = req.working_directory {
         workdir_path.join(working_directory)
       } else {
         workdir_path.to_owned()
       })
-      .envs(&req.env)
-      .stream(&req)
+      .envs(&req.env);
+    let _locked = if exclusive_spawn {
```

Comment:
Would you mind adding a comment here pointing to #10507 or to https://github.com/golang/go/issues/22315 ?

It is not crystal clear to me how this fixes the issue, as my read of the go ticket is that another thread in this process would need to be writing _the same_ file?

Reply:
The semantics of `tokio::process:Command::spawn` are a synchronous spawn:
```
...snip...
/// This method will spawn the child process synchronously and return a
/// handle to a future-aware child process
...snip...
```

As such - when spawn returns, the fork/exec is guaranteed completed.

In short, by grabbing the write lock, we're guaranteed to have all prior forks (caused by this method) that may have grabbed a copy of our open fd to have completed their exec and thus run their CLOEXECs. We're also guaranteed no concurrent forks.

Reply:
Ok, I think I see it. Adding a comment to that effect would be great.

Reply:
I'll add a comment and pointers. It looks like you understood the issue on second take, so adding a sentence or two that explains it on one take (for anyone familiar with unix fork/exec + threads share process memory space semantics) would be good. I'll also highlight that this only works while we have exactly one place in the scheduler code that does fork/exec.

Reply:
Fixed.